### ISSUE TITLE
crash using `gb` with single-line file

### ISSUE THREAD

**Issue by johncalab** [helix-editor/helix#1205]
possibly related to: https://github.com/helix-editor/helix/issues/934

### Reproduction steps
My `config.toml` contains only the line `"theme" = "bogster"`
Steps:

```
hx config.toml
gb
```

output

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Line index out of bounds: line index 5, Rope/RopeSlice line count 1', /home/johncalab/.cargo/registry/src/github.com-1ecc6299db9ec823/ropey-1.3.1/src/rope.rs:766:41
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Environment

- Platform: Linux
- Helix version: helix v0.5.0-171-gc08d2fa

<details><summary>~/.cache/helix/helix.log</summary>
```
2021-11-30T16:14:43.373 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { LSP not defined }
```

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

**Comment by johncalab** [comment]
I just installed helix from master a couple hours ago.

**Comment by archseer** [comment]
\cc @QiBaobin this seems to be caused by the `goto_window` changes. Both `gt`/`gb` panic

**Comment by archseer** [comment]
```
.max(view.offset.row + scrolloff);
```

seems to be unnecessary. https://github.com/helix-editor/helix/commit/42fde95223a62bded340a1737e8be50ef94af4af#diff-d782c07f22492bcae17333aba5dc5b840fb30317727e71267ebf4863f84787fdR755

**Comment by archseer** [comment]
Closed by #1207
